README-pt-br.md

```diff
@@ -0,0 +1,1730 @@
+```js
+// Atribuindo uma função para a constante foo
+const foo = function() {};
+```
+
+# 👀 Exemplos
+
+## `[]` é igual a `![]`
+
+Array é igual a not array:
+
+```js
+[] == ![]; // -> true
+```
+
+### 💡 Explicação:
+
+O operador abstrato de igualdade converde os dois lados em números para compará-los, e os dois lados se tornam `0` por razões diferentes. Arrays são verdadeiros (truthy), então na direita, o oposto de um valor verdadeiro é `false`, o que é coagido para `0`. Na esquerda, todavia, um array vazio é coagido para um número sem se tornar um booleano (boolean) primeiro, e arrays vazios sempre forçados para `0`, apesar de serem verdadeiros.
+
+Aqui está uma simplificação dessa expressão:
+
+```js
++[] == +![];
+0 == +false;
+0 == 0;
+true;
+```
+
+Veja também [`[]` is truthy, but not `true`](#-is-truthy-but-not-true).
+
+- [**12.5.9** Logical NOT Operator (`!`)](https://www.ecma-international.org/ecma-262/#sec-logical-not-operator)
+- [**7.2.13** Abstract Equality Comparison](https://www.ecma-international.org/ecma-262/#sec-abstract-equality-comparison)
+
+## `true` não é igual a `![]`, nem igual a `[]` também
+
+Array não é igual a `true`, mas not Array também não é igual a `true`'
+Array é igual a `false`, not Array é igual a `false` também:
+
+```js
+true == []; // -> false
+true == ![]; // -> false
+
+false == []; // -> true
+false == ![]; // -> true
+```
+
+### 💡 Explicação:
+
+```js
+true == []; // -> false
+true == ![]; // -> false
+
+// De acordo com a especificação
+
+true == []; // -> false
+
+toNumber(true); // -> 1
+toNumber([]); // -> 0
+
+1 == 0; // -> false
+
+true == ![]; // -> false
+
+![]; // -> false
+
+true == false; // -> false
+```
+
+```js
+false == []; // -> true
+false == ![]; // -> true
+
+// De acordo com a especificação
+
+false == []; // -> true
+
+toNumber(false); // -> 0
+toNumber([]); // -> 0
+
+0 == 0; // -> true
+
+false == ![]; // -> true
+
+![]; // -> false
+
+false == false; // -> true
+```
+
+- [**7.2.13** Abstract Equality Comparison](https://www.ecma-international.org/ecma-262/#sec-abstract-equality-comparison)
+
+## true é false
+
+```js
+!!"false" == !!"true"; // -> true
+!!"false" === !!"true"; // -> true
+```
+
+### 💡 Explicação:
+
+Considere esse passo-a-passo:
+
+```js
+// true é 'truthy' e representado pelo valor 1 (number), 'true' como string é NaN.
+true == "true"; // -> false
+false == "false"; // -> false
+
+// 'false' não é uma string vazia, então ele é um valor verdadeiro (truthy)
+!!"false"; // -> true
+!!"true"; // -> true
+```
+
+- [**7.2.13** Abstract Equality Comparison](https://www.ecma-international.org/ecma-262/#sec-abstract-equality-comparison)
+
+## baNaNa
+
+```js
+"b" + "a" + +"a" + "a"; // -> 'baNaNa'
+```
+
+Essa é uma piada antiga no JavaScript, mas remasterizada. Aqui está a forma original:
+
+```js
+"foo" + +"bar"; // -> 'fooNaN'
+```
+
+### 💡 Explicação:
+
+A expressão é evaluada como `'foo' + (+'bar')`, o que converte `bar` para um "não número" (NaN - Not a Number).
```

Comment:
evaluada ou avaliada?

Reply:
I'd stick to "avaliada" since "evaluada" isn't a Portuguese word (it's Spanish), but I've been away for a while and I'not sure if it's been adopted somehow as a dev jargon. Your call.

Reply:
Agree, it's hard to translate some words as we adopt and get them as "natural", however, I'd stick with avaliada.

Reply:
I've always used _evaluada_, but I think it's a dev jargon (as @tbalmeida) said; I'll change it to _avaliada_, since it's the correct translation.